pkg/artifactory/resource_artifactory_watch.go

```diff
@@ -0,0 +1,654 @@
+	if len(allRepositoriesRaw) == 1 {
+		params.Repositories.Type = xrayutils.WatchRepositoriesAll
+
+		if allRepositoriesRaw[0] != nil {
+			allRepositories := allRepositoriesRaw[0].(map[string]interface{})
+			names := castToStringArr(allRepositories["names"].(*schema.Set).List())
+			sort.Strings(names)
+
+			packageTypes := castToStringArr(allRepositories["package_types"].(*schema.Set).List())
+			sort.Strings(packageTypes)
+			paths := castToStringArr(allRepositories["paths"].(*schema.Set).List())
+			sort.Strings(paths)
+			mimeTypes := castToStringArr(allRepositories["mime_types"].(*schema.Set).List())
+			sort.Strings(mimeTypes)
+
+			params.Repositories.All.Filters.Names = names
+			params.Repositories.All.Filters.PackageTypes = packageTypes
+			params.Repositories.All.Filters.Paths = paths
+			params.Repositories.All.Filters.MimeTypes = mimeTypes
+
+			params.Repositories.All.Filters.Properties = map[string]string{}
+			// Properties are nested in another level
+			for _, rawValue := range allRepositories["property"].(*schema.Set).List() {
+				value := rawValue.(map[string]interface{})
+				params.Repositories.All.Filters.Properties[value["key"].(string)] = value["value"].(string)
+			}
+		}
+	}
+}
+
+func unpackRepository(repositoriesRaw []interface{}, artClient *artifactoryold.Artifactory, params *xrayutils.WatchParams) error {
+	if len(repositoriesRaw) > 0 {
+		params.Repositories.Type = xrayutils.WatchRepositoriesByName
+		// There can be 1 to M repositories
+		params.Repositories.Repositories = make(map[string]xrayutils.WatchRepository, 0)
+
+		for _, repositoryRaw := range repositoriesRaw {
+
+			watchRepo := xrayutils.WatchRepository{}
+			repository := repositoryRaw.(map[string]interface{})
+			name := repository["name"].(string)
+			binMgrID := repository["bin_mgr_id"].(string)
+			repoType := repository["repo_type"].(string)
+
+			// GetLocal will retrieve any repo type by name.
+			repo, _, err := artClient.V1.Repositories.GetLocal(context.Background(), name)
+			if err != nil {
+				return fmt.Errorf("error getting %s repo %s %v", repoType, name, err)
+			}
+
+			// A watch requires the repo to have xray indexing on.
+			if repo != nil {
+				if *repo.XrayIndex == false {
+					return fmt.Errorf("repo %s must have xray indexing on", name)
+				}
+			}
+
+			if *repo.RClass != repoType {
+				return fmt.Errorf("found repo %s with %s type, but %s was specified", name, *repo.RClass, repoType)
+			}
+
+			watchRepo.Name = name
+			watchRepo.BinMgrID = binMgrID
+			switch repoType {
+			case "remote":
+				watchRepo.RepoType = xrayutils.WatchRepositoryRemote
+			case "local":
+				watchRepo.RepoType = xrayutils.WatchRepositoryLocal
+			}
+
+			packageTypes := castToStringArr(repository["package_types"].(*schema.Set).List())
+			sort.Strings(packageTypes)
+			paths := castToStringArr(repository["paths"].(*schema.Set).List())
+			sort.Strings(paths)
+			mimeTypes := castToStringArr(repository["mime_types"].(*schema.Set).List())
```

Comment:
Why are you sorting these?

And, please put spaces between the string slices and the sort pair

Reply:
I'm sorting the lists to ensure that terraform does not think there is a change if the returned list is ordered differently.

Reply:
Pesky sets/hashes!